r1cs/gadgets/crypto/src/crh/poseidon/mod.rs

```diff
@@ -71,8 +71,11 @@ impl<
 
             // Add the round constants to the state vector
             for d in state.iter_mut() {
-                let rc = P::ROUND_CST[round_cst_idx];
-                (*d).add_constant_in_place(cs.ns(|| format!("add_constant_1_{}", round_cst_idx)), &rc)?;
+                let rc = FpGadget::<ConstraintF>::from_value(
+                    cs.ns(|| format!("hardcode round constant {}", round_cst_idx)),
+                    &P::ROUND_CST[round_cst_idx]
+                );
+                *d = rc.add(cs.ns(|| format!("add_constant_{}", round_cst_idx)), d)?;
```

Comment:
What again is the reason for that different procedure? 
Has it to do with stopping propagation of LCs?

Reply:
Yes

Reply:
We could comment it as "temporary work-around".